testsuite/manualmode/src/test/java/org/wildfly/core/test/standalone/mgmt/PreparedResponseTestCase.java

```diff
@@ -0,0 +1,152 @@
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ManagementClient;
+import org.wildfly.core.testrunner.ServerControl;
+import org.wildfly.core.testrunner.ServerController;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+import org.wildfly.test.shutdown.SlowStopService;
+import org.wildfly.test.shutdown.SlowStopServiceActivator;
+
+/**
+ *
+ * @author <a href="mailto:ehugonne@redhat.com">Emmanuel Hugonnet</a> (c) 2015 Red Hat, inc.
+ */
+@RunWith(WildflyTestRunner.class)
+@ServerControl(manual = true)
+public class PreparedResponseTestCase {
+
+    public static Logger LOGGER = Logger.getLogger(PreparedResponseTestCase.class);
+
+    public static final String SLOW_STOP_JAR = "slow-stop.jar";
+    private static final long FREQUENCY = 500L;
+
+    @Inject
+    protected static ServerController controller;
+
+    @BeforeClass
+    public static void startAndSetupContainer() throws Exception {
+        controller.start();
+        ManagementClient managementClient = controller.getClient();
+        deploy(managementClient);
+    }
+
+    @AfterClass
+    public static void stopContainer() throws Exception {
+        undeploy(controller.getClient());
+        controller.stop();
+    }
+
+    public static void deploy(ManagementClient managementClient) throws Exception {
+        ServerDeploymentHelper helper = new ServerDeploymentHelper(managementClient.getControllerClient());
+        JavaArchive war = ShrinkWrap.create(JavaArchive.class, SLOW_STOP_JAR);
+        war.addPackage(SlowStopService.class.getPackage());
+        war.addAsServiceProvider(ServiceActivator.class, SlowStopServiceActivator.class);
+        war.addAsResource(new StringAsset("Dependencies: org.jboss.dmr, org.jboss.logging, org.jboss.as.controller, org.jboss.as.server,org.wildfly.extension.request-controller, org.jboss.as.network\n"), "META-INF/MANIFEST.MF");
+        helper.deploy(SLOW_STOP_JAR, war.as(ZipExporter.class).exportAsInputStream());
+    }
+
+    public static void undeploy(ManagementClient managementClient) throws ServerDeploymentHelper.ServerDeploymentException {
+        ServerDeploymentHelper helper = new ServerDeploymentHelper(managementClient.getControllerClient());
+        helper.undeploy(SLOW_STOP_JAR);
+    }
+
+    @Test
+    public void reloadServer() throws Exception {
+        ManagementClient mgmtClient = getManagementClient();
+        long timeout = SHUTDOWN_WAITING_TIME + System.currentTimeMillis();
+        mgmtClient.executeForResult(Operations.createOperation(RELOAD, new ModelNode().setEmptyList()));
+        while (System.currentTimeMillis() < timeout) {
+            Thread.sleep(FREQUENCY);
+            try {
+                controller.getClient().isServerInRunningState();
+            } catch (RuntimeException ex) {
+                ex.printStackTrace();
+                break;
+            }
+        }
+        Assert.assertFalse(System.currentTimeMillis() < timeout);
+        Thread.sleep(2 * FREQUENCY);
```

Comment:
This should use the `org.jboss.as.test.shared.TimeoutUtil.adjust()` to calculate the sleep time.